microfactory/microfactory.go

```diff
@@ -69,6 +69,7 @@ var (
 
 	goToolDir = filepath.Join(runtime.GOROOT(), "pkg", "tool", runtime.GOOS+"_"+runtime.GOARCH)
 	goVersion = findGoVersion()
+	isGo18    = strings.Contains(goVersion, "go1.8")
 )
 
 func findGoVersion() string {
```

Comment:
This function seems inaccurate.  If you can assume that microfactory was built with the go version in GOROOT, you can use runtime.Version(), but for the beta won't all 3 of these methods return a SHA and not a string?